File: backend/app/Modules/Vehiculo/Models/Vehiculo.php
<?php

namespace App\Modules\Vehiculo\Models;

use App\Models\User;
use App\Modules\Vehiculo\Enums\EstadoVehiculo;
use App\Modules\Vehiculo\Enums\TipoEnergia;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;

class Vehiculo extends Model
{
    protected $table = 'vehiculos';

    protected $fillable = [
        'ficha',
        'placa',
        'marca',
        'modelo',
        'anio',
        'tipo_energia',
        'capacidad_carga',
        'estado',
        'created_by',
    ];

    protected $casts = [
        'estado'          => EstadoVehiculo::class,
        'tipo_energia'    => TipoEnergia::class,
        'capacidad_carga' => 'decimal:2',
        'anio'            => 'integer',
    ];

    /**
     * Mapea el campo PHP "anio" al campo real de la BD "año".
     * Se usa esta convencion para evitar problemas con caracteres UTF-8 en nombre de columna.
     */
    public function getAnioAttribute(): ?int
    {
        return $this->attributes['año'] ?? null;
    }

    public function setAnioAttribute(?int $value): void
    {
        $this->attributes['año'] = $value;
    }

    // --- Relaciones ---

    public function creador(): BelongsTo
    {
        return $this->belongsTo(User::class, 'created_by');
    }

    public function mantenimientos(): HasMany
    {
        return $this->hasMany(VehiculoMantenimiento::class, 'vehiculo_id');
    }

    public function gastos(): HasMany
    {
        return $this->hasMany(VehiculoGasto::class, 'vehiculo_id');
    }

    // --- Helpers ---

    public function estaDisponible(): bool
    {
        return $this->estado === EstadoVehiculo::Disponible;
    }

    public function estaEnMantenimiento(): bool
    {
        return $this->estado === EstadoVehiculo::EnMantenimiento;
    }
}

File: frontend/lib/modules/flota/models/vehiculo.dart
class Vehiculo {
  final int id;
  final String ficha;
  final String? placa;
  final String? marca;
  final String? modelo;
  final int? anio;
  final String tipoEnergia;
  final double? capacidadCarga;
  final String estado;

  Vehiculo({
    required this.id,
    required this.ficha,
    this.placa,
    this.marca,
    this.modelo,
    this.anio,
    required this.tipoEnergia,
    this.capacidadCarga,
    required this.estado,
  });

  factory Vehiculo.fromJson(Map<String, dynamic> json) {
    return Vehiculo(
      id: json['id'],
      ficha: json['ficha'] ?? '',
      placa: json['placa'],
      marca: json['marca'],
      modelo: json['modelo'],
      anio: json['anio'],
      tipoEnergia: json['tipo_energia'] ?? 'diesel',
      capacidadCarga: json['capacidad_carga'] != null 
          ? double.tryParse(json['capacidad_carga'].toString()) 
          : null,
      estado: json['estado'] ?? 'disponible',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'ficha': ficha,
      'placa': placa,
      'marca': marca,
      'modelo': modelo,
      'anio': anio,
      'tipo_energia': tipoEnergia,
      'capacidad_carga': capacidadCarga,
      'estado': estado,
    };
  }
}